Data/ByteString.hs

```diff
@@ -1679,7 +1679,9 @@ hPutNonBlocking h bs@(PS ps s l) = do
 hPutStr :: Handle -> ByteString -> IO ()
 hPutStr = hPut
 
--- | Write a ByteString to a handle, appending a newline byte
+-- | Write a ByteString to a handle, appending a newline byte.
+-- Another thread might write to the handle after it wrote the
+-- bytestring and before the newline was written.
```

Comment:
It's a bit unclear what "it" refers to. Is it that other thread?

I'd also suggest to mention "non-atomic" – IMHO that's a keyword that will give a good context for many users.

Reply:
I rephrased it, added it to putStrLn and added the newline.

cabal haddock fails for me locally so for now that has to do.